src/constraint/field.ts

```diff
@@ -90,15 +90,15 @@ export const FIELD_CONSTRAINTS: EncodingConstraintModel<FieldQuery>[] = [
         ...toFieldDef(fieldQ, {schema, props: ['bin', 'timeUnit', 'type']})
       };
 
-      const {compatible} = channelCompatibility(fieldDef, fieldQ.channel as Channel);
+      const {compatible} = channelCompatibility(fieldDef, fieldQ.channel as ExtendedChannel);
 
       if (compatible) {
         return true;
       } else {
         // In VL, facet's field def must be discrete (O/N), but in CompassQL we can relax this a bit.
         const isFacet = fieldQ.channel === 'row' || fieldQ.channel === 'column';
 
-        if (isFacet && (isLocalSingleTimeUnit(fieldDef.timeUnit) || isUtcSingleTimeUnit(fieldDef.timeUnit))) {
+        if (isFacet && !!fieldDef.timeUnit && (isLocalSingleTimeUnit(fieldDef.timeUnit as string) || isUTCTimeUnit(fieldDef.timeUnit as string))) {
```

Comment:
This will break when time unit is not a string. I guess this cannot be the case since we control what we generate.

Reply:
I'm not sure how to take a Typescript type and turn it into a string to match the function definition of `isLocalSingleTimeUnit` from Vega-Lite. It seems that maybe there should be a separate set of functions to take TimeUnit as input rather than strings as input in Vega-Lite's time unit functionality?

Reply:
Can you normalize the time unit with `normalizeTimeUnit(timeUnit)?.unit`? We do something like that in https://github.com/vega/vega-lite/blob/master/src/channeldef.ts#L1322. 

isLocalSingleTimeUnit only supports strings.

Reply:
Yes it works thank you!